util/registry/test/test_filelike.py

```diff
@@ -149,3 +149,12 @@ def test_slice_explictread():
     assert stream.read(2) == b"is"
     assert stream.read(5) == b" a"
     assert len(b"is a") == stream.tell()
+
+
+def test_non_filelike_obj():
+    from gunicorn.http.body import Body
+
+    fileobj = BytesIO(b"this will not really be a real fileobj")
+    body = Body(fileobj)
+    ls = LimitingStream(body)
+    assert ls.readable()
```

Comment:
Execute a read as well, and ensure it finds the correct data and is limited

Reply:
Done